pkg/util/node/node.go

```diff
@@ -143,6 +143,20 @@ func GetZoneKey(node *v1.Node) string {
 	return region + ":\x00:" + failureDomain
 }
 
+// GetNodeCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetNodeCondition(status *v1.NodeStatus, conditionType v1.NodeConditionType) (int, *v1.NodeCondition) {
```

Comment:
This is only used by controllers it seems so maybe `pkg/controller/util/node` is a better place?

Reply:
@andrewsykim `GetNodeCondition` is also called in e2e test.
Eg:  https://github.com/kubernetes/kubernetes/pull/73656/files#diff-4efe416092e6fea48c89ed2144d7825bR126

Reply:
This code is so trivial it would be better to split it into one for e2e and one for controllers.

Reply:
@smarterclayton  @andrewsykim  does [this commit](https://github.com/kubernetes/kubernetes/pull/73656/commits/e6592408af95065dbfb7209c8a45a0e13db01f0b) looks good ?